src/Service/UserRequestService.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\UserRequest;
+use App\Form\Model\Search;
+use App\Repository\UserRequestRepositoryInterface;
+use App\Traits\UserAwareTrait;
+use Doctrine\Common\Collections\Criteria;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+class UserRequestService
+{
+    use UserAwareTrait;
+
+    public function __construct(
+        private readonly AuthorizationCheckerInterface $authorizationChecker,
+        private readonly Security $security,
+    ) {
+    }
+
+    /** @return UserRequest[] */
+    public function getAccessibleUserRequests(Search $search, UserRequestRepositoryInterface $repository): array
+    {
+        $parameters = ['done' => $search->getShowDone() ?? false];
+
+        $qb = $repository->createQueryBuilder('ur')
+            ->leftJoin('ur.user', 'u')
+            ->leftJoin('ur.application', 'a')
+            ->andWhere('ur.done = :done')
+            ->addOrderBy('ur.done', Criteria::ASC)
+            ->addOrderBy('ur.createdAt', Criteria::DESC);
```

Comment:
We could add these criteria in the entity I guess

Reply:
Not sure i understand what you mean